### ISSUE TITLE
[prefer-readonly-parameter-types] IndexSignature internals don't violate the rule

### ISSUE THREAD

**Issue by RebeccaStevens** [typescript-eslint/typescript-eslint#3714]
I wasn't sure if this was intentional behavior or not so I thought I may as well open an issue on it.

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/prefer-readonly-parameter-type": "error"
  }
}
```

```TS
type Foo = {
  readonly [key: string]: {
    groups: string[];
  };
};

function foo(arg: Foo) {}
```

**Expected Result**

The `PropertySignature` in the `IndexSignature` is not readonly and thus the rule should be violated.

**Actual Result**

No rule violation.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.29.0` |
| `@typescript-eslint/parser`        | `4.29.0` |
| `TypeScript`                       | `4.3.5` |
| `ESLint`                           | `7.27.0` |
| `node`                             | `16.4.0` |

**Comment by kmin-jeong** [comment]
can I try it?
